src/pseudopeople/configuration/noise_configuration.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Optional, Union
+
+from layered_config_tree import LayeredConfigTree
+
+from pseudopeople.entity_types import ColumnNoiseType, NoiseType, RowNoiseType
+from pseudopeople.noise_entities import NOISE_TYPES
+from pseudopeople.schema_entities import DATASET_SCHEMAS
+
+COLUMN_NOISE_TYPES = [
+    noise_type.name for noise_type in NOISE_TYPES if isinstance(noise_type, ColumnNoiseType)
+]
+ROW_NOISE_TYPES = [
+    noise_type.name for noise_type in NOISE_TYPES if isinstance(noise_type, RowNoiseType)
+]
+
+
+class NoiseConfiguration:
+    def __init__(self, config: LayeredConfigTree):
+        self._config = config
+
+    def to_dict(self) -> dict:
+        config_dict: dict = self._config.to_dict()
+        return config_dict
+
+    def get_parameter_value(
+        self,
+        dataset: str,
+        noise_type: str,
+        column_name: Optional[str] = None,
+        parameter_name: Optional[str] = None,
+    ) -> Union[float, int]:
+        config = self._config
+        try:
+            dataset_config = config[dataset]
+        except:
+            raise ValueError(
+                f"{dataset} was not found in configuration. "
+                f"Available datasets are {list(config.keys())}"
+            )
+
+        def _get_value_without_parameter_name(
+            config: LayeredConfigTree, noise_type: str
+        ) -> Union[int, float]:
+            parameters: LayeredConfigTree = config[noise_type]
+            if len(parameters) > 1:
+                available_parameters = list(parameters.keys())
+                raise ValueError(
+                    f"Your noising configuration has multiple parameters for the {dataset} dataset and noise type {noise_type}, so you must provide a parameter. "
+                    f"The available parameters are {available_parameters}"
+                )
+            else:
+                parameter_name = list(parameters.keys())[0]
+                value: Union[int, float] = config[noise_type][parameter_name]
+                return value
+
+        # row noise
+        if noise_type in ROW_NOISE_TYPES:
```

Comment:
You have duplicated code here, since dealing with row and columns noise is identical once you get to the correct depth in the config tree. Both look like this:

```python

# validate inputs
...
# get config subtree
config = ...

if parameter_name not in config:
    raise ValueError(
        f"The parameter {parameter_name} was not found for {noise_type} in the configuration. "
        f"Available parameters are {list(row_parameters.keys())}."
    )

return config[parameter_name]
```